### DISCUSSION TITLE
strictNullChecks=false makes z.object() properties optional

### DISCUSSION THREAD

**Discussion by AndrewSmir** [colinhacks/zod#2160] [Q&A] [answered]
Hi!

Turning off strictNullChecks in compilerOptions makes all object properties optional.

tsc -v: 4.9.5

tsconfig:
```ts
{
  "compilerOptions": {
    "strictNullChecks": false,
    "strict": true,
  }
}
```
**Example:**

```ts
const Dog = z.object({
  name: z.string(),
  age: z.number(),
});

type Dog = z.infer<typeof Dog>;
```

**Expected result:**
// equivalent to:
```ts
type Dog = {
  name: string;
  age: number;
};
```

**Actual result:**
// equivalent to:
```ts
type Dog = {
  name?: string;
  age?: number;
};
```

**Comment by JacobWeisenburger** [comment, marked as answer]
https://github.com/colinhacks/zod#requirements
> You must enable `strict` mode in your `tsconfig.json`. This is a best practice for all TypeScript projects.

if you disable `strictNullChecks`, then you are disabling part of `strict` mode which is required for zod to work correctly.
